packages/shared/lib/services/getting-started.service.ts

```diff
@@ -0,0 +1,288 @@
+import { pick } from 'lodash-es';
+
+import db from '@nangohq/database';
+import { getProvider } from '@nangohq/providers';
+import { Err, Ok } from '@nangohq/utils';
+
+import configService from './config.service.js';
+import connectionService from './connection.service.js';
+import sharedCredentialsService from './shared-credentials.service.js';
+
+import type {
+    CreateGettingStartedMeta,
+    CreateGettingStartedProgress,
+    DBConnection,
+    DBEnvironment,
+    DBGettingStartedMeta,
+    DBGettingStartedProgress,
+    DBUser,
+    GettingStartedOutput as GettingStartedProgressOutput,
+    IntegrationConfig,
+    PatchGettingStartedInput
+} from '@nangohq/types';
+import type { Result } from '@nangohq/utils';
+import type { Knex } from 'knex';
+
+export async function createMeta(input: CreateGettingStartedMeta, trx?: Knex.Transaction): Promise<Result<DBGettingStartedMeta>> {
```

Comment:
I'll set a default value for the optional parameter
```
function createMeta( db: knex.Knex, input: CreateGettingStartedProgress) ...
```

For consistency I will make the db param first and add it to all functions that talk to the db, even the one that are not used in a transaction, so if one day they must be called from within a transaction we don't forget.

I would also make it a required param. The caller will always have a db instance to pass